Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java

```diff
@@ -0,0 +1,266 @@
+
+    public static List<ServiceResourceInner> listAllSpringCloudClusters() throws IOException {
+        final List<ServiceResourceInner> clusters = new ArrayList<>();
+        List<Subscription> subs = AzureMvpModel.getInstance().getSelectedSubscriptions();
+        if (subs.size() == 0) {
+            return clusters;
+        }
+        Observable.from(subs).flatMap((sd) -> Observable.create((subscriber) -> {
+            try {
+                List<ServiceResourceInner> clustersInSubs = listAllSpringCloudClustersBySubscription(
+                        sd.subscriptionId());
+                synchronized (clusters) {
+                    clusters.addAll(clustersInSubs);
+                }
+            } catch (IOException e) {
+                // swallow exception and skip error subscription
+            }
+            subscriber.onCompleted();
+        }).subscribeOn(Schedulers.io()), subs.size()).subscribeOn(Schedulers.io()).toBlocking().subscribe();
+        return clusters;
+    }
+
+    public static List<ServiceResourceInner> listAllSpringCloudClustersBySubscription(String sid) throws IOException {
+        PagedList<ServiceResourceInner> res = getSpringManager(sid).inner().services().list();
+        res.loadAll();
+        return res;
+    }
+
+    public static DeploymentResourceInner getAppDeployment(String sid, String resourceGroupName, String clusterName,
+            String appName, String deploymentName) throws IOException {
+        return getSpringManager(sid).deployments().inner().get(resourceGroupName, clusterName, appName, deploymentName);
+    }
+
+    public static DeploymentResourceInner getAppDeployment(String appId, String deploymentName) throws IOException {
+        String sid = getSubscriptionId(appId);
+        String rid = getResourceGroup(appId);
+        String cid = getClusterName(appId);
+        String appName = getAppName(appId);
+        return getSpringManager(sid).deployments().inner().get(rid, cid, appName, deploymentName);
+    }
+
+    public static AppResourceInner getAppByName(String sid, String resourceGroupName, String clusterName,
+            String appName) throws IOException {
+        return getSpringManager(sid).apps().inner().get(resourceGroupName, clusterName, appName);
+    }
+
+    public static AppResourceInner getAppById(String appId) throws IOException {
+        return getSpringManager(getSubscriptionId(appId)).apps().inner().get(getResourceGroup(appId),
+                getClusterName(appId), getAppName(appId));
+    }
+
+    public static DeploymentResourceInner getActiveDeploymentForApp(String appId) throws IOException {
+        String sid = getSubscriptionId(appId);
+        String rid = getResourceGroup(appId);
+        String cid = getClusterName(appId);
+        String appName = getAppName(appId);
+        AppResourceInner app = getSpringManager(sid).apps().inner().get(rid, cid, appName);
```

Comment:
use `getAppByName` and `getAppDeployment` here

Reply:
Done